pkg/velero/backup.go

```diff
@@ -0,0 +1,353 @@
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the backup is empty")
+
+		builder.errorMsg = "backup namespace cannot be an empty string"
+	}
+
+	return builder
+}
+
+// PullBackup loads an existing backup into BackupBuilder struct.
+func PullBackup(apiClient *clients.Settings, name, nsname string) (*BackupBuilder, error) {
+	glog.V(100).Infof("Pulling existing backup name: %s under namespace: %s", name, nsname)
+
+	builder := BackupBuilder{
+		apiClient: apiClient.VeleroClient,
+		Definition: &velerov1.Backup{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		return nil, fmt.Errorf("backup name cannot be empty")
+	}
+
+	if nsname == "" {
+		return nil, fmt.Errorf("backup namespace cannot be empty")
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("backup object %s doesn't exist in namespace %s", name, nsname)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// WithStorageLocation adds a storage location to the backup.
+func (builder *BackupBuilder) WithStorageLocation(location string) *BackupBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Adding storage location %s to backup %s in namespace %s",
+		location, builder.Definition.Name, builder.Definition.Namespace)
+
+	if location == "" {
+		glog.V(100).Infof("Backup storage location is empty")
+
+		builder.errorMsg = "backup storage location cannot be an empty string"
+	}
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	if builder.Definition.ObjectMeta.Labels == nil {
+		builder.Definition.ObjectMeta.Labels = make(map[string]string)
+	}
+
+	builder.Definition.ObjectMeta.Labels["velero.io/storage-location"] = location
+
+	return builder
+}
+
+// WithIncludedNamespace adds the specified namespace for inclusion when performing a backup.
+func (builder *BackupBuilder) WithIncludedNamespace(namespace string) *BackupBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Adding namespace %s to backup %s in namespace %s included namespaces",
+		namespace, builder.Definition.Name, builder.Definition.Namespace)
+
+	if namespace == "" {
+		glog.V(100).Infof("Backup included namespace is empty")
+
+		builder.errorMsg = "backup included namespace cannot be an empty string"
+	}
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	builder.Definition.Spec.IncludedNamespaces = append(builder.Definition.Spec.IncludedNamespaces, namespace)
+
+	return builder
+}
+
+// WithIncludedClusterScopedResource adds the specified cluster-scoped crd for inclusion when performing a backup.
+func (builder *BackupBuilder) WithIncludedClusterScopedResource(crd string) *BackupBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Adding customresource %s to backup %s in namespace %s included cluster-scoped resources",
```

Comment:
should included be change to including?

Reply:
The resource is called IncludedClusterScopedResources